lib/ansible/modules/cloud/vmware/vmware_vm_storage_policy_info.py

```diff
@@ -0,0 +1,164 @@
+  for virtual machines and storage capabilities of storage providers.
+version_added: '2.9'
+author:
+- Abhijeet Kasurde (@Akasurde)
+notes:
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.6
+- PyVmomi
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Get SPBM info
+  vmware_vm_storage_policy_info:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    validate_certs: no
+  delegate_to: localhost
+  register: profiles
+'''
+
+RETURN = r'''
+spbm_profiles:
+  description: list of dictionary of SPBM facts
+  returned: success
+  type: list
+  sample: [
+        {
+            "constraints_sub_profiles": [
+                {
+                    "rule_set_info": [
+                        {
+                            "id": "hostFailuresToTolerate",
+                            "value": 1
+                        },
+                        {
+                            "id": "stripeWidth",
+                            "value": 1
+                        },
+                        {
+                            "id": "forceProvisioning",
+                            "value": false
+                        },
+                        {
+                            "id": "proportionalCapacity",
+                            "value": 0
+                        },
+                        {
+                            "id": "cacheReservation",
+                            "value": 0
+                        }
+                    ],
+                    "rule_set_name": "VSAN sub-profile"
+                }
+            ],
+            "description": "Storage policy used as default for vSAN datastores",
+            "id": "aa6d5a82-1c88-45da-85d3-3d74b91a5bad",
+            "name": "vSAN Default Storage Policy"
+        },
+    ]
+'''
+
+try:
+    from pyVmomi import pbm
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware_spbm import SPBM
+from ansible.module_utils.vmware import vmware_argument_spec
+
+
+class SPBMClient(SPBM):
+    def __init__(self, module):
+        super(SPBMClient, self).__init__(module)
+
+    def show_capabilities(self, capabilities):
+        """
+        Return property instance for given capabilities
+        """
+        capabilities_info = []
+        for capability in capabilities:
+            for constraint in capability.constraint:
+                if hasattr(constraint, 'propertyInstance'):
+                    for propertyInstance in constraint.propertyInstance:
+                        capabilities_info.append(
+                            {
+                                'id': propertyInstance.id,
+                                'value': propertyInstance.value
+                            }
+                        )
+        return capabilities_info
+
+    def ensure(self):
```

Comment:
Method name doesnt give much info about what it does. is it possible to change?

Thanks

Reply:
OK will change to `get_policy_info` or something.